src/frontend/skipToken.sk

```diff
@@ -151,15 +138,29 @@ fun escapeStringLiteralValue(s: String): String {
   s.chars().map(escapeCharLiteralValue).join("")
 }
 
-fun accumulateStringValue(acc: String, value: String, index: Int): String {
-  if (index == value.length()) {
-    acc
-  } else {
-    length = charLiteralLength(value, index);
-    accumulateStringValue(
-      acc + charLiteral(value.sub(index, length)),
-      value,
-      index + length,
-    )
-  }
+fun stringValue(value: String): String {
+  chars = value.chars();
+  acc = mutable Vector[];
+  i = 0;
+  while (i < chars.size()) {
+    literal = mutable Vector[];
+    literalSize = chars[i] match {
+    | '\\' ->
+      chars[i + 1] match {
+      | 'x' -> 4
+      | 'u' -> 6
+      | 'U' -> 10
+      | _ -> 2
+      }
+    | _ -> 1
+    };
+    for (j in Range(0, literalSize)) {
```

Comment:
```
limit = Int.compare(chars.size() - i, literalSize) match {
| LT() -> chars.size() - i
| _ -> literalSize
};
for j in Range(0, limit) {
  literal.push(chars[i + j])
}
```
(Is there a more succinct way to write the `limit` initialization using some existing comparison function?)

Reply:
```js
limit = if (chars.size() - i < literalSize) chars.size() - i else literalSize
```
?

Reply:
Or
```js
limit = min(chars.size() - i, literalSize)
```

Reply:
Ah cool, I found `Int.min`, but not `min()`.